cpp/marc_tools/marc_filter.cc

```diff
@@ -627,16 +629,28 @@ bool ReplaceSubfields(const std::vector<std::string> &subfield_specs, RegexMatch
 
         if (modified_at_least_one_subfield) {
             modified_at_least_one_field = true;
-            field.setContents(std::string(1, field.getIndicator1()) + std::string(1, field.getIndicator2()) + subfields.toString());
+            field->setContents(std::string(1, field->getIndicator1()) + std::string(1, field->getIndicator2())
+                              + subfields.toString());
+            if (unlikely(field->empty())) {
+                indices_of_deleted_fields.emplace_back(field - record->begin());
+                tags_of_deleted_fields.insert(field->getTag().toString());
+            }
         }
     }
 
+    // Did we generate completely empty fields?
+    if (unlikely(not indices_of_deleted_fields.empty())) {
+        LOG_WARNING("regex \"" + matcher.getPattern() + "\" led to empty fields in the record w/ control number "
+                    + record->getControlNumber() + " and field(s) " + StringUtil::Join(tags_of_deleted_fields, ',') + "!");
+        record->deleteFields(indices_of_deleted_fields);
```

Comment:
Just a a note: An operation on subfields might lead to invalidating an iterator on the fields of the record